lib/Sema/TypeCheckType.cpp

```diff
@@ -4291,19 +4291,22 @@ TypeResolver::resolveOwnershipTypeRepr(OwnershipTypeRepr *repr,
       options.hasBase(TypeResolverContext::SubscriptDecl) ||
       options.hasBase(TypeResolverContext::EnumElementDecl)) {
 
-    decltype(diag::attr_only_on_parameters) diagID;
-    if (options.getBaseContext() == TypeResolverContext::SubscriptDecl) {
-      diagID = diag::attr_not_on_subscript_parameters;
-    } else if (options.is(TypeResolverContext::VariadicFunctionInput)) {
-      diagID = diag::attr_not_on_variadic_parameters;
-    } else {
-      diagID = diag::attr_only_on_parameters;
-    }
     StringRef name;
     if (ownershipRepr) {
       name = ownershipRepr->getSpecifierSpelling();
     }
-    diagnoseInvalid(repr, repr->getSpecifierLoc(), diagID, name);
+
+    if (options.hasBase(TypeResolverContext::SubscriptDecl) ||
+        options.is(TypeResolverContext::VariadicFunctionInput) ||
```

Comment:
We used to produce a tailored diagnostic for "variadic" parameters and I think we should keep doing that because they are allowed in functions and initializers. Using the new diagnostic you added would be confusing in such cases because the issue is not with the parent declaration but with the parameter itself.

Reply:
Should also add a couple of tests that uses func/init with variable and `inout`.

Reply:
I see, I reverted the specific diagnostic for variadic parameters and added tests accordingly.